test/mitmproxy/addons/test_proxyserver.py

```diff
@@ -290,7 +290,7 @@ async def test_dns(caplog_async) -> None:
         ps.running()
         await caplog_async.await_log("DNS server listening at")
         assert ps.servers
-        dns_addr = ps.servers["dns@127.0.0.1:0"].listen_addrs[0]
+        dns_addr = ps.servers["dns@127.0.0.1:0"].listen_addrs[1]
```

Comment:
Why is this necessary? Are `listen_addrs[0]` and `listen_addrs[1]` not the same?

Reply:
`test_udp_*` tests fail with the earlier version of the code as there's no response returned from the server. I suspect `listen_addrs[0]` returns the address of the TCP server here and `listen_addrs[1]` is the UDP server's address

Reply:
Thanks. I think the point here is that TCP and UDP should listen on the same port if possible. We probably need to extend this workaround here to also apply to UDP:

https://github.com/mitmproxy/mitmproxy/blob/40ae9ff05b3a9abe2f919f40c9ca4319ce8a62ca/mitmproxy/proxy/mode_servers.py#L275-L284